src/selectors/entities/schemes.js

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2018-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+// @flow
+
+import {getAllChannels} from 'selectors/entities/channels';
+import {getTeams} from 'selectors/entities/teams';
+import {ScopeTypes} from 'constants/schemes';
+import type {GlobalState} from 'types/store';
+import type {Scheme} from 'types/schemes';
+import type {Channel} from 'types/channels';
+import type {Team} from 'types/teams';
+
+export function getSchemes(state: GlobalState): { [string]: Scheme } {
+    return state.entities.schemes.schemes;
+}
+
+export function getScheme(state: GlobalState, id: string): Scheme {
+    const schemes = getSchemes(state);
+    return schemes[id];
+}
+
+export function getSchemeChannels(state: GlobalState, schemeId: string): Array<Channel> {
+    const {scope} = getScheme(state, schemeId);
+
+    if (scope === ScopeTypes.TEAM) {
+        throw new Error(`Not implemented: scheme '${schemeId}' is team-scope but 'getSchemeChannels' only accepts channel-scoped schemes.`);
```

Comment:
Does anyone have any opinions about throwing an error here? I'm 0/5 about whether this rather returns an empty array in this case.

Reply:
Yeah I think it's totally valid to just return an empty array instead of throwing the error and I'd suggest doing that instead 👍

Reply:
@jwilander Maybe I'll return an empty array and print a JS warning so that a developer using it realizes that it's not a valid param to pass a `schemeId` associated to the wrong scope.